src/include/artifact.h

```diff
@@ -63,6 +63,7 @@ typedef struct mender_artifact_checksum_t mender_artifact_checksum_t;
 struct mender_artifact_checksum_t {
     char                       *filename;
     unsigned char               manifest[MENDER_DIGEST_BUFFER_SIZE];
+    bool                        checked;
```

Comment:
An alternative is to actually drop the checked checksums from the list. I'd prefer that because we don't need them after being checked and it's quite some heap space to get back.

Reply:
@vpodzime I did this on a separate commit to ease your review.